### PR TITLE
fix: make emoji pick more accessible 

### LINKED ISSUE #2411: Bug: Emoji picker menu items are inaccessible via keyboard
### Describe the bug

Emoji picker menu items are inaccessible via keyboard

### Steps to reproduce

1. Ensure you're logged in to the https://app.opensauced.pizza.
2. Go to the highlights feed, https://app.opensauced.pizza/feed
3. Click on the smiley icon then press tab
4. Press <kbd>Shift</kbd> + <kbd>TAB</kbd>, and the smiley icon gets focus (a good thing)
5. Press <kbd>ENTER</kbd>. The emoji picker menu opens.
6. Press <kbd>TAB</kbd>.
7. Notice that the first emoji in the list does not receive focus. The focus goes to the next focusable element, a link.

CleanShot 2023-12-29 at 13 11 36

### Browsers

_No response_

### Additional context (Is this in dev or production?)

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### Contributing Docs

- [X] I agree to follow this project's Contribution Docs

### PR DESCRIPTION
**Pull request by CBID2**
## Description
This PR makes the emoji buttons more accessible, creating are user-friendly experience for individuals who use screen readers to navigate the internet. So far,  @nickytonline and I added a hover state.   
**Update:** @Arol15 and I tested to see if the tabbing works. From there, we omit the `<div>` elements and replace them with `<button>` tags, which makes the code more semantic and easier for screen readers to scan. 

## What type of PR is this? (check all applicable)

- [X] 🍕 Feature
- [X] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI
- [ ] 📦 Chore (Release)
- [ ] ⏩ Revert

## Related Tickets & Documents

Closes #2411 
### Before

https://github.com/open-sauced/app/assets/105683440/6c906165-4b05-4eac-af78-6ad754c28be1

### After

https://github.com/open-sauced/app/assets/105683440/9d123c13-63eb-4788-8758-224d8c1b8739

## Steps to QA

1. Visit https://deploy-preview-2412--oss-insights.netlify.app/feed
2. Validate accessibility has improved 

## Added to documentation?

- [ ] 📜 README.md
- [ ] 📓 docs.opensauced.pizza
- [ ] 🍕 dev.to/opensauced
- [ ] 📕 storybook
- [x] 🙅 no documentation needed

### CODE REVIEW

**Comment by Arol15** [comment]
I'm trying to review Christine's PR locally but I'm having a hard time checking it out locally.

1. I forked the repo
2. I cloned it locally
3. Configured the remote repo for my fork. 
4. Followed the instructions on how to run it locally - runs with no issues. 
5. Ran `git fetch upstream` to fetch the branches and their respective commits from the upstream repository.
6. Ran `git fetch upstream pull/2412/head:CBID2:making-emoji-picker-accessible`
7. Was getting an error: fatal: invalid refspec 'pull/2412/head:CBID2:making-emoji-picker-accessible'

NOTE: I was able to checkout this PR with gh CLI and noticed that the branch's name was `making-emoji-picker-accessible` and not `CBID2:making-emoji-picker-accessible`. However, it's not listed in the list of all branches for some reason. Also, by running `git fetch upstream pull/2412/head:making-emoji-picker-accessible` I successfully pulled this PR and could see the changes. 

Can someone please explain why I wasn't able to find this branch in the list of all branches in the repo and also when I ran `git branch -r`?

**Comment by bdougie** [comment]
> I'm trying to review Christine's PR locally but I'm having a hard time checking it out locally.

FWIW - You can get there in less steps with the [GitHub CLI](https://cli.github.com/). In your fork run the following.  

```
gh pr checkout 2412
```

You of course need to install the CLI: `brew install gh` on mac

**Comment by bdougie** [comment]
> I'm trying to review Christine's PR locally but I'm having a hard time checking it out locally.

@CBID2 it would be helpful for reviewers to include screenshots and some steps to QA this fix.

**Comment by CBID2** [comment]
> > I'm trying to review Christine's PR locally but I'm having a hard time checking it out locally.
> 
> 
> 
> @CBID2 it would be helpful for reviewers to include screenshots and some steps to QA this fix. 

Hey @bdougie. I tried running it locally too, but it didn't appear on my screen

**Comment by Arol15** [comment]
> FWIW - You can get there in less steps with the [GitHub CLI](https://cli.github.com/). In your fork run the following.
Hey @bdougie! Yes, I did that eventually but also like to be able to run the same commands using just git. Thank you! :)

**Comment by bdougie** [comment]
> error: fatal: invalid refspec 'pull/2412/head:CBID2:making-emoji-picker-accessible'

Looking at the error on the computer now and that is because you most likely need to set up a remote that points to the fork `CBID2/app`, but I am assuming you are running this locally from your fork `Arol15/app`. Without seeing what is in your `git remote -v` list, this is just my first guess. 

The gh cli clones, forks, and sets the remotes up for you in one command.

**Comment by Arol15** [comment]
> > > I'm trying to review Christine's PR locally but I'm having a hard time checking it out locally.
> > 
> > 
> > @CBID2 it would be helpful for reviewers to include screenshots and some steps to QA this fix.
> 
> Hey @bdougie. I tried running it locally too, but it didn't appear on my screen

I followed @nickytonline instructions listed in the issue https://github.com/open-sauced/app/issues/2411 and it seems that the changes made didn't fix the issue. 

> > error: fatal: invalid refspec 'pull/2412/head:CBID2:making-emoji-picker-accessible'
> 
> Looking at the error on the computer now and that is because you most likely need to set up a remote that points to the fork `CBID2/app`, but I am assuming you are running this locally from your fork `Arol15/app`. Without seeing what is in your `git remote -v` list, this is just my first guess.
> 
> The gh cli clones, forks, and sets the remotes up for you in one command.

Ah, I see! 
Screenshot of the output of git remote -v command
Yes, you are right. I just realized that it's on Christine's fork and hasn't been merged into open-sauced. Thank YOU! I will use gh cli :)

**Comment by bdougie** [comment]
@CBID2 I approved the build in Netlify so there is a deploy preview. Also updated the Steps to QA. Outside of lighthouse scores and [testing in React](https://dev.to/bdougieyo/accessibility-testing-in-react-with-jest-axe-l7k). Not sure if there is a specific way to test this outside of confirming the page is not broken, which it is not. 


https://deploy-preview-2412--oss-insights.netlify.app/feed?redirectedFrom=%2F

**Comment by CBID2** [comment]
> @CBID2 I approved the build in Netlify so there is a deploy preview. Also updated the Steps to QA. Outside of lighthouse scores and [testing in React](https://dev.to/bdougieyo/accessibility-testing-in-react-with-jest-axe-l7k). Not sure if there is a specific way to test this outside of confirming the page is not broken, which it is not. 
> 
> 
> 
> 
> 
> https://deploy-preview-2412--oss-insights.netlify.app/feed?redirectedFrom=%2F

Thanks @bdougie! :) @Arol15, you can test it now

**Comment by Arol15** [comment]
I followed @nickytonline instructions listed in the issue https://github.com/open-sauced/app/issues/2411 and it seems that the changes made didn't fix the issue. 



> > @CBID2 I approved the build in Netlify so there is a deploy preview. Also updated the Steps to QA. Outside of lighthouse scores and [testing in React](https://dev.to/bdougieyo/accessibility-testing-in-react-with-jest-axe-l7k). Not sure if there is a specific way to test this outside of confirming the page is not broken, which it is not.
> > https://deploy-preview-2412--oss-insights.netlify.app/feed?redirectedFrom=%2F
> 
> Thanks @bdougie! :) @Arol15, you can test it now

I don't think I'm seeing the changes we are expecting... According to @nickytonline 's instructions, pressing tab should go to the eyes emoji (in my case) and then to the link. But it's not happening. See the video below. 

I think it's a separate issue but I can't use my keyboard to pick an emoji from the menu either. 

https://github.com/open-sauced/app/assets/25575320/b4317178-ee40-4ebb-a636-a6d0c8bd8c00

**Comment by bdougie** [comment]
> I think it's a separate issue but I can't use my keyboard to pick an emoji from the menu either.

 I see it works with the up and down arrows.

**Comment by Arol15** [comment]
> > I think it's a separate issue but I can't use my keyboard to pick an emoji from the menu either.
> 
> I see it works with the up and down arrows.

Oops!! I tried tab and right and left arrows only 🙈 I even tested the behavior with github emoji menu and the right and left worked for me. I guess both combinations should work (but i'm def not an expert in this).

**Comment by Arol15** [comment]
@CBID2 @bdougie FYI, I'm trying to figure out the issue.

**Comment by CBID2** [comment]
> > I think it's a separate issue but I can't use my keyboard to pick an emoji from the menu either.
> 
> I see it works with the up and down arrows.

@CBID2 and @Arol15,
here's my results


https://github.com/open-sauced/app/assets/105683440/3675916f-743b-4e3f-9262-73d9dc3d8ea7

**Comment by Arol15** [comment]
> @CBID2 and @Arol15, here's my results. It works with the arrows too.
> 
Left and right? Or up and down?

**Review by nickytonline — CHANGES_REQUESTED** [review]
The CSS changes you made @CBID2 are correct, but the JSX needs to be changed to buttons, so this becomes accessible. Currently they are divs. See https://github.com/open-sauced/app/issues/2411#issuecomment-1872258746

**Comment by Arol15** [comment]
Hey @bdougie! I have a question for future reference. If I ever need to push commits to someone’s PR, what steps should I follow? When I committed my changes to @CBID2  ’s PR, I attempted to run `git push --set-upstream git@github.com:CBID2/app-my-version.git making-emoji-picker-accessible` since my remote repository is `open-sauced/app`. Initially, I encountered permission issues, but after gaining access to the PR author's repository, I received the error message: `error: failed to push some refs to 'github.com:CBID2/app-my-version.git'`. I tried using `git pull` and `git fetch` commands, but unfortunately, they didn't resolve the issue. It seems like I might be missing a crucial step, especially when working with multiple forks simultaneously. Your guidance would be greatly appreciated!

**Comment by CBID2** [comment]
> Hey @bdougie! I have a question for future reference. If I ever need to push commits to someone’s PR, what steps should I follow? When I committed my changes to @CBID2  ’s PR, I attempted to run `git push --set-upstream git@github.com:CBID2/app-my-version.git making-emoji-picker-accessible` since my remote repository is `open-sauced/app`. Initially, I encountered permission issues, but after gaining access to the PR author's repository, I received the error message: `error: failed to push some refs to 'github.com:CBID2/app-my-version.git'`. I tried using `git pull` and `git fetch` commands, but unfortunately, they didn't resolve the issue. It seems like I might be missing a crucial step, especially when working with multiple forks simultaneously. Your guidance would be greatly appreciated!

We managed to push the commits @bdougie. It was some technical difficulties on my end

**Comment by bdougie** [comment]
> Hey @bdougie! I have a question for future reference.

It would be easier to take this question to the [discord](https://discord.com/invite/opensauced). I believe @RitaDee wrote a blog on this back in August, and I think the community benefit from this discussion as well.

**Review by OgDev-01 — APPROVED** [review]
Works as espected after swithching the `div` to `button` 👏👏

**Comment by Arol15** [comment]
> > Hey @bdougie! I have a question for future reference.
> 
> It would be easier to take this question to the [discord](https://discord.com/invite/opensauced). I believe @RitaDee wrote a blog on this back in August, and I think the community benefit from this discussion as well.

Will do! thank you!

**Comment by CBID2** [comment]
> Works as espected after swithching the `div` to `button` 👏👏

Hey @bdougie. Nich does not come back till the 8th. If you're free, can you give this PR a second review?

**Comment by nickytonline** [comment]
> > Works as espected after swithching the `div` to `button` 👏👏
> 
> Hey @bdougie. Nich does not come back till the 8th. If you're free, can you give this PR a second review?

I'm going to take a peek at this on Monday.

In the PR description, please add a video of before where it is inaccessible and one for after where it's accessible.

Thanks!

**Comment by CBID2** [comment]
> > > Works as espected after swithching the `div` to `button` 👏👏
> > 
> > 
> > Hey @bdougie. Nich does not come back till the 8th. If you're free, can you give this PR a second review?
> 
> I'm going to take a peek at this on Monday.
> 
> In the PR description, please add a video of before where it is inaccessible and one for after where it's accessible.
> 
> Thanks!

@Arol15 and I added the videos @nickytonline! :) See you on Monday! :)

**Comment by nickytonline** [comment]
The build is currently failing with this error. Looks like syntax errors in CSS. I'm going to put it back into draft mode until the build errors are sorted.

```
5:29:16 PM: <s> [webpack.Progress] 99% cache begin idle
5:29:16 PM: <s> [webpack.Progress] 100%
5:29:16 PM: Module build failed (from ./node_modules/@storybook/addon-styling/node_modules/postcss-loader/dist/cjs.js):
5:29:16 PM: 
5:29:16 PM: SyntaxError
5:29:16 PM: 
5:29:16 PM: (29:1) ./styles/globals.css Unclosed block
5:29:16 PM: 
5:29:16 PM:   27 |   @apply border-transparent;
5:29:16 PM:   28 | }
5:29:16 PM: > 29 | input[type="checkbox"]:focus-visible,
5:29:16 PM:      | ^
5:29:16 PM:   30 | [role="checkbox"]:focus-visible,
5:29:16 PM:   31 | [role="radio"]:focus-visible,
```

**Comment by nickytonline** [comment]
> > Hey @CBID2, just following up on this. Are you still working on this?
> 
> I'm done @nickytonline. It needs your review

@CBID2, the build does not build as mentioned in https://github.com/open-sauced/app/pull/2412#issuecomment-1891090349

There are also file conflicts that need to be sorted out.

**Comment by CBID2** [comment]
> > > Hey @CBID2, just following up on this. Are you still working on this?
> 
> > 
> 
> > I'm done @nickytonline. It needs your review
> 
> 
> 
> @CBID2, the build does not build as mentioned in https://github.com/open-sauced/app/pull/2412#issuecomment-1891090349
> 
> 
> 
> There are also file conflicts that need to be sorted out.

Fixed the syntax error @nickytonline

File: components/molecules/ContributorHighlight/contributor-highlight-card.tsx [hunk 1]
```diff
@@ -667,7 +667,12 @@ const ContributorHighlightCard = ({
                   <Emoji className="text-xs text-orange-500 md:text-sm" text={`:${getEmojiNameById(emoji_id)}:`} />
                   <span>{reaction_count}</span>
                 </span>
-              </button>
+        </button>
+                <Emoji
+                  className="text-xs text-orange-500 md:text-sm"
+                  text={`:${getEmojiNameById(emoji_id)}: ${reaction_count}`}
+                />
+              </div>
```
Comment (Arol15) [inline components/molecules/ContributorHighlight/contributor-highlight-card.tsx:675]: @CBID2 we need to remove this. i think this came from the merge conflict commit.

Comment (CBID2) [reply components/molecules/ContributorHighlight/contributor-highlight-card.tsx]: Good idea @Arol15. You have my permission to do this :)

Comment (Arol15) [reply components/molecules/ContributorHighlight/contributor-highlight-card.tsx]: hmm, i think the above lines (667-669) are the old code. I'll double-check. don't remove it yet.

File: styles/globals.css [hunk 2]
```diff
@@ -22,12 +22,16 @@ h5,
...
+button:not([role]){
+  @apply border;
+  @apply border-transparent;
+}
+input[type="checkbox"]:focus-visible;
+[role="checkbox"]:focus-visible;
+[role="radio"]:focus-visible;
+[role="combobox"]:focus-visible;
+[role="switch"]:focus-visible;
+button:not([role]):focus-visible{
```
Comment (Arol15) [inline styles/globals.css:34]: remove duplicate 
```suggestion


```

File: components/molecules/ContributorHighlight/contributor-highlight-card.tsx [hunk 3]
```diff
@@ -657,12 +657,17 @@ const ContributorHighlightCard = ({
...
+              <button
                 className={`px-1 py-0 md:py-0.5 hover:bg-light-slate-6 transition  md:px-1.5 shrink-0 border flex items-center justify-center rounded-full cursor-pointer ${
                   isUserReaction(emoji_id) && "bg-light-slate-6"
                 }`}
                 onClick={async () => (sessionToken ? handleUpdateReaction(emoji_id) : signIn({ provider: "github" }))}
               >
+                <span className="flex gap-1 items-center p-1">
+                  <Emoji className="text-xs text-orange-500 md:text-sm" text={`:${getEmojiNameById(emoji_id)}:`} />
+                  <span>{reaction_count}</span>
+                </span>
```
Comment (Arol15) [inline components/molecules/ContributorHighlight/contributor-highlight-card.tsx:669]: remove these lines 
```suggestion

```

File: components/molecules/ContributorHighlight/contributor-highlight-card.tsx [hunk 4]
```diff
@@ -657,12 +657,13 @@ const ContributorHighlightCard = ({
...
               content={`${getEmojiReactors(reaction_users)} reacted with ${getEmojiNameById(emoji_id)} emoji`}
             >
-              <div
+              <button
                 className={`px-1 py-0 md:py-0.5 hover:bg-light-slate-6 transition  md:px-1.5 shrink-0 border flex items-center justify-center rounded-full cursor-pointer ${
                   isUserReaction(emoji_id) && "bg-light-slate-6"
                 }`}
                 onClick={async () => (sessionToken ? handleUpdateReaction(emoji_id) : signIn({ provider: "github" }))}
               >
+        </button>
```
Comment (Arol15) [inline components/molecules/ContributorHighlight/contributor-highlight-card.tsx:666]: this button should go to line 671 and replace `</div` right after closing tag for `Emoji` component. I can't use `make a suggestion` button bc that line was added before the latest change.

Comment (CBID2) [reply components/molecules/ContributorHighlight/contributor-highlight-card.tsx:666]: Thanks @Arol15! 😊 I'll try this tomorrow

Comment (Arol15) [inline components/molecules/ContributorHighlight/contributor-highlight-card.tsx:666]: hey @CBID2! You just need to remove this `</button>` AFTER you replace `</div> `with `</button>` on line `671`.

Comment (Arol15) [reply components/molecules/ContributorHighlight/contributor-highlight-card.tsx:666]:
```suggestion

```

**Comment by CBID2** [comment]
Hey @nickytonline! :) I made the changes and things are ok :)

**Comment by BekahHW** [comment]
@CBID2 this will need updated with the beta branch before it can be merged in.

**Comment by CBID2** [comment]
> @CBID2 this will need updated with the beta branch before it can be merged in.

Done @BekahHW

**Review by brandonroberts — APPROVED** [review]

File: styles/globals.css [hunk 5]
```diff
@@ -22,12 +22,15 @@ h5,
...
-
-input[type="checkbox"]:focus-visible,
-[role="checkbox"]:focus-visible,
-[role="radio"]:focus-visible,
-[role="combobox"]:focus-visible,
-[role="switch"]:focus-visible,
+button:not([role]){
+  @apply border;
+  @apply border-transparent;
+}
```
Comment (nickytonline) [inline styles/globals.css:28]: nit: add spacing here
```suggestion
}

```

**Review by nickytonline — APPROVED** [review]
Thanks for the PR!

There's more that can be done here, but this makes it more accesible thanks to the trigger for the menu being a button now.

Some good things to add in a subsequent PR would be pressing Escape to exit the open menu, and maybe make the emoji menu items buttons as well so you can tab through them.

**Comment by CBID2** [comment]
> Thanks for the PR!
> 
> 
> 
> There's more that can be done here, but this makes it more accesible thanks to the trigger for the menu being a button now.
> 
> 
> 
> Some good things to add in a subsequent PR would be pressing Escape to exit the open menu, and maybe make the emoji menu items buttons as well so you can tab through them.

Good idea @nickytonline! 😊 I'll raise this as an issue! :)

**Pull request merged**